### ISSUE TITLE
Wrong diagnostics range for readability-duplicate-include

### ISSUE THREAD

**Issue by timxx** [clangd/clangd#1125]
Take the [official example](https://clang.llvm.org/extra/clang-tidy/checks/readability-duplicate-include.html)  for example:
```cpp
#include <memory>
#include <vector>
#include <memory>
```

Enable `readability-duplicate-include` from config.yaml `Diagnostics/ClangTidy/Add` section

clangd will reports two positions (with unwanted previous line) for `readability-duplicate-include`, see the following verbose:
`V[14:02:31.713] >>> {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[{"code":"readability-duplicate-include","message":"Duplicate include (fix available)","range":{"end":{"character":17,"line":2},"start":{"character":17,"line":1}},"relatedInformation":[],"severity":2,"source":"clang-tidy"}],"uri":"file:///test.cpp","version":1}}
`

It work just fine with clang-tidy.

**System information**

Output of `clangd --version`: 14.0.1

Editor/LSP plugin: VS Code/clangd

Operating system: Both Windows & Linux

**Comment by HighCommander4** [comment]
Agreed, this seems to be an awkward way of representing "the entire line 2" (line numbers being 0-based in LSP) as a range. The start of the range is `{"character":17,"line":1}`, which is the last position on the previous line, but I don't see any reason why it couldn't be `{"character":0,"line":2}` instead.
